From b70a011ee0c31166df16cff871da706235fe5d12 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 09 Feb 2020 10:52:22 +0200
Subject: [PATCH] [SW-8229] goya2: if key hash is valid use it

If hash compare of image public key matches the first
tested hash key stored in security we can exit for loop
and continue testing.

Change-Id: I43f460948b1577fe2d895fdaa3923d7334cacb6b
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/secureboot.c b/common/secureboot.c
index 230c02b..6a275f6 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -69,7 +69,9 @@
 						     &verify, block_buf,
 						     sizeof(block_buf));
 		}
-		/* TODO: if customer key is ok exit for */
+		/* if key is verified we can exit the loop */
+		if (verify)
+			break;
 	}
 	/* if for loop doesn't find a valid key hash in fuse, return error */
 	if (!verify && (key_scan == SECUREBOOT_HABANA_KEY)) {
